\section{AbuseHumanDB [NOT SOLVED]}

donc a priori il y a dans la bdd des entrée et celle qui contient le flag n'est
pas retournée car son statut n'est pas appouved.

pas de SQLi a priori



il y a un bot. Il visite une url et c'est tout a priori

la question c'est comment une entrée est crée en bdd déjà

par ce que ce n'est 


quand on cherche les entrées

\begin{verbatim}
router.get('/api/entries/search', (req, res) => {
	if(req.query.q) {
		const query = `${req.query.q}%`;
		return db.getEntry(query, isLocalhost(req))
			.then(entries => {
				if(entries.length == 0) return res.status(404).send(response('Your search did not yield any results!'));
				res.json(entries);
			})
			.catch(() => res.send(response('Something went wrong! Please try again!')));
	}
	return res.status(403).json(response('Missing required parameters!'));
});
\end{verbatim}

\begin{verbatim}
const isLocalhost = req => ((req.ip == '127.0.0.1' && req.headers.host == '127.0.0.1:1337') ? 0 : 1);
\end{verbatim}

\begin{verbatim}
async getEntry(query, approved=1) {
        return new Promise(async (resolve, reject) => {
            try {
                let stmt = await this.db.prepare("SELECT * FROM userEntries WHERE title LIKE ? AND approved = ?");
                resolve(await stmt.all(query, approved));
            } catch(e) {
                console.log(e);
                reject(e);
            }
        });
    }
\end{verbatim}

donc il faut que la recheche soit faite en local 
